numpy/_core/tests/test_strings.py

```diff
@@ -796,6 +796,54 @@ def test_zfill(self, buf, width, res, dt):
         assert_array_equal(np.strings.zfill(buf, width), res)
 
 
+@pytest.mark.parametrize("dt", [
+    "U",
+    "S",
+    pytest.param("T", marks=pytest.mark.xfail(
+        reason="StringDType support not implemented",
+        strict=True),
+    ),
+])
+class TestMethodsWithoutStringDTypeSupport:
+    @pytest.mark.parametrize("buf,sep,res1,res2,res3", [
+        ("this is the partition method", "ti", "this is the par",
+            "ti", "tion method"),
+        ("http://www.python.org", "://", "http", "://", "www.python.org"),
```

Comment:
How about including a bit of a test of broadcasting by doing
```
("http://www.python.org", ["://", "?", "http://", "org"], [...
```
Could also be the mississippi one if that keeps things a bit clearer.

(and the same for the other tests)